src/plugins/graph/utilities/graph-utils.ts

```diff
@@ -140,16 +145,34 @@ export interface IMatchCirclesProps {
 }
 
 export function matchCirclesToData(props: IMatchCirclesProps) {
-
   const {dataConfiguration, enableAnimation, instanceId,
-      dotsElement, pointRadius, pointColor, pointStrokeColor} = props,
-    allCaseData = dataConfiguration.joinedCaseDataArrays,
-    caseDataKeyFunc = (d: CaseData) => `${d.plotNum}-${d.caseID}`,
-    circles = selectCircles(dotsElement);
-  if (!circles) return;
+      dotsElement, pointRadius, pointColor, pointStrokeColor} = props;
+  const allCaseData = dataConfiguration.joinedCaseDataArrays;
+
+  // Remove the circles if they already existed
+  // We needed to do this because initializing the circles without first removing them resulted in one set of
+  // circles not being created. We weren't able to figure out why this was happening.
+  let allCircles = selectAllCircles(dotsElement); //includes both inner and outer circles
+  if (!allCircles) return;
+  allCircles.remove();
```

Comment:
You shouldn't need to remove all of the circles and then recreate them -- that's going to be a performance problem. I suspect that issue is that you don't have an `update` function for the outer highlight dots. Note that for the inner dots the `join` is passed an `enter` function, which determines how new dots are created, and an `update` function, which determines how existing dots are updated. Passing an appropriate `update` function for the outer dots should allow you to skip this initial `remove()`.

Reply:
I just tried adding an update function to the outer dots and it didn't fix the problem. It's an issue when you first create a graph--graphs that are already a part of the document when it loads seem to be fine, and series added after graph creation are fine. There's other curious behavior here that might point to what's going on, though I wasn't able to make sense of it to figure it out.
- The order of the circles determines which set fails to show up. So if you initialize the inner circles first, then the outer circles, it's the outer circles that fail to materialize. Whichever is later fails.
- It's always the first set of dots that gets screwed up. So if you add a second series, it's fine. Then if you remove the first series, the second series switches to its place and is messed up.